diff --git a/cdk/cmake/DepFindSSL.cmake b/cdk/cmake/DepFindSSL.cmake
index 4ec374aa..3680f83d 100644
--- a/cdk/cmake/DepFindSSL.cmake
+++ b/cdk/cmake/DepFindSSL.cmake
@@ -120,7 +120,9 @@ function(main)
 
     set_target_properties(openssl-applink PROPERTIES FOLDER "Misc")
     # Remove warnings from openssl applink.c
-    target_compile_options(openssl-applink PRIVATE /wd4152 /wd4996)
+    if (MSVC)
+	    target_compile_options(openssl-applink PRIVATE /wd4152 /wd4996)
+    endif()
 
   endif()
 
diff --git a/cmake/libutils/merge_archives.cmake.in b/cmake/libutils/merge_archives.cmake.in
index eebbdb73..e110f483 100644
--- a/cmake/libutils/merge_archives.cmake.in
+++ b/cmake/libutils/merge_archives.cmake.in
@@ -319,10 +319,16 @@ function(merge_libraries_gcc)
   #
   # TODO: Do we need any linking options to pass here?
 
+  list(JOIN merge_objs " " joined_objs)
+  file(WRITE objects ${joined_objs})
   execute_process(
-    COMMAND ${CMAKE_AR} -r ${TARGET} ${merge_objs}
+    COMMAND cat objects
+    COMMAND xargs ${CMAKE_AR} -r ${TARGET}
     RESULT_VARIABLE res
   )
+  if (res)
+    message(FATAL_ERROR "Failed AR")
+  endif()
   execute_process(
     COMMAND ${CMAKE_RANLIB} ${TARGET}
     RESULT_VARIABLE res1
